datahub-web-react/src/app/entity/domain/DataProductsTab/CreateDataProductModal.tsx

```diff
@@ -32,6 +32,7 @@ export default function CreateDataProductModal({ domain, onCreateDataProduct, on
             variables: {
                 input: {
                     domainUrn: domain.urn,
+                    id: builderState.id || '',
```

Comment:
not  || ''

this will cause an empty string id to be created. 
let's just say builderState.id.

Undefined | null should be provided if an id is not

Reply:
Done